onnx/defs/tensor/defs.cc

```diff
@@ -1121,4 +1121,29 @@ ONNX_OPERATOR_SET_SCHEMA(
             OpSchema::all_tensor_types(),
             "Constrain input and output types to all tensor types.")
         .TypeAndShapeInferenceFunction(propagateShapeAndTypeFromFirstInput));
+
+static const char* Compress_ver1_doc = R"DOC(
+    Compress a tensor that behaves like numpy.compress: https://docs.scipy.org/doc/numpy/reference/generated/numpy.compress.html
+    )DOC";
+ONNX_OPERATOR_SET_SCHEMA(
+    Compress,
+    9,
+    OpSchema()
+        .SetDoc(Compress_ver1_doc)
+        .Attr(
+            "axis",
+            "Axis along which to take slices.",
+            AttributeProto::INT,
+            static_cast<int64_t>(0))
```

Comment:
If we want to align with Numpy, we should not set default value as 0

Reply:
Gather takes indices. Compress takes condition of bools as input tensor. condition input may naturally come from ops like logical operations. It is not possible to use Gather to achieve what is needed here.

Reply:
regarding default axis, I agree we shall follow numpy's default behavior - if not set, assume flatten source.

Reply:
Make sense.

Reply:
How about the default value of axis?

Reply:
I am thinking there is no default value for axis attribute. So if the axis attribute is not set, the op will perform as stated: to flatten the input and to select from the flattened tensor according to the condition evaluation. The output, in this case, is a rank 1 tensor. I will add an example for this case.